### PR TITLE
PROJQUAY-85 - Travis CI: Lint for Python syntax errors

### PR DESCRIPTION
**Pull request by cclauss**
### Description of Changes

* details about the implementation of the changes
* motivation for the change (broken code, new feature, etc)
* contrast with previous behavior

#### Changes:

* ..
* ..

#### Issue: https://issues.redhat.com/browse/PROJQUAY-85

**TESTING** ->

**BREAKING CHANGE** ->

---

## Reviewer Checklist

- [ ] It works!
- [ ] Comments provide sufficient explanations for the next contributor
- [ ] Tests cover changes and corner cases
- [ ] Follows Quay syntax patterns and format

### CODE REVIEW

**Comment by thomasmckay** [comment]
This is great, thanks! We're going to do a mass change to PEP-8 python formatting so we'll wait to test and merge.

If you're willing, creating an issue to track this at https://issues.jboss.org/projects/PROJQUAY/issues would be great. (If you do not wish to do this, we'll create one on your behalf. Frequent contributors are expected to do this step, however.)

There are three commits in this PR, please squash them into one unless there is a reason to keep unique.

Welcome!

**Comment by cclauss** [comment]
Given that these tests pass is there any reason to wait?

File: .travis.yml [hunk 1]
```diff
@@ -24,14 +22,26 @@ cache:
...
 # We should label the steps if Travis ever supports it:
 # https://github.com/travis-ci/travis-ci/issues/5898
 jobs:
+  allow_failures:
+    - python: 3.8
   include:
+    - stage: lint
+      script:
+        - pip install flake8
+        - flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
```
Comment (thomasmckay) [inline .travis.yml]: Mind changing this to a config file?

What was the reason to confine the select to these four types?

Comment (cclauss) [reply .travis.yml]: https://flake8.pycqa.org/en/latest/user/error-codes.html

On the flake8 test selection, this PR does _not_ focus on "_style violations_" (the majority of flake8 error codes that [__psf/black__](https://github.com/psf/black) can autocorrect).  Instead these tests are focus on runtime safety and correctness:
* E9 tests are about Python syntax errors usually raised because flake8 can not build an Abstract Syntax Tree (AST).  Often these issues are a sign of unused code or code that has not been ported to Python 3.  These would be compile-time errors in a compiled language but in a dynamic language like Python they result in the script halting/crashing on the user.
* F63 tests are usually about the confusion between identity and equality in Python.  Use ==/!= to compare str, bytes, and int literals is the classic case.  These are areas where __a == b__ is True but __a is b__ is False (or vice versa).  Python >= 3.8 will raise SyntaxWarnings on these instances.
* F7 tests logic errors and syntax errors in type hints
* F82 tests are almost always _undefined names_ which are usually a sign of a typo, missing imports, or code that has not been ported to Python 3.  These also would be compile-time errors in a compiled language but in Python a __NameError__ is raised which will halt/crash the script on the user.

**Comment by cclauss** [comment]
Closing in favor of #124.
